python/cuml/preprocessing/model_selection.py

```diff
@@ -13,469 +13,12 @@
-
-    if shuffle:
-        # Shuffle the data
-        if random_state is None or isinstance(random_state, int):
-            idxs = cp.arange(X.shape[0])
-            random_state = cp.random.RandomState(seed=random_state)
-
-        elif isinstance(random_state, cp.random.RandomState):
-            idxs = cp.arange(X.shape[0])
-
-        elif isinstance(random_state, np.random.RandomState):
-            idxs = np.arange(X.shape[0])
-
-        else:
-            raise TypeError("`random_state` must be an int, NumPy RandomState \
-                             or CuPy RandomState.")
-
-        random_state.shuffle(idxs)
-
-        if isinstance(X, cudf.DataFrame) or isinstance(X, cudf.Series):
-            X = X.iloc[idxs]
-
-        elif hasattr(X, "__cuda_array_interface__"):
-            # numba (and therefore rmm device_array) does not support
-            # fancy indexing
-            X = cp.asarray(X)[idxs]
-
-        if isinstance(y, cudf.DataFrame) or isinstance(y, cudf.Series):
-            y = y.iloc[idxs]
-
-        elif hasattr(y, "__cuda_array_interface__"):
-            y = cp.asarray(y)[idxs]
-
-        if stratify is not None:
-            split_return = _stratify_split(X,
-                                           y,
-                                           train_size,
-                                           test_size,
-                                           x_numba,
-                                           y_numba,
-                                           random_state)
-            return split_return
-
-    # If not stratified, perform train_test_split splicing
-    if hasattr(X, "__cuda_array_interface__"):
-        x_order = _strides_to_order(X.__cuda_array_interface__['strides'],
-                                    cp.dtype(X.dtype))
-
-    if hasattr(y, "__cuda_array_interface__"):
-        y_order = _strides_to_order(y.__cuda_array_interface__['strides'],
-                                    cp.dtype(y.dtype))
-
-    if hasattr(X, "__cuda_array_interface__") or \
-            isinstance(X, cupyx.scipy.sparse.csr_matrix):
-        X_train = cp.array(X[0:train_size], order=x_order)
-        if y is not None:
-            y_train = cp.array(y[0:train_size], order=y_order)
-    elif isinstance(X, cudf.DataFrame):
-        X_train = X.iloc[0:train_size]
-        if y is not None:
-            y_train = y.iloc[0:train_size]
-
-    if hasattr(X, "__cuda_array_interface__") or \
-            isinstance(X, cupyx.scipy.sparse.csr_matrix):
-        X_test = cp.array(X[-1 * test_size:], order=x_order)
-        if y is not None:
-            y_test = cp.array(y[-1 * test_size:], order=y_order)
-    elif isinstance(X, cudf.DataFrame):
-        X_test = X.iloc[-1 * test_size:]
-        if y is not None:
-            y_test = y.iloc[-1 * test_size:]
-    if x_numba:
-        X_train = cuda.as_cuda_array(X_train)
-        X_test = cuda.as_cuda_array(X_test)
+from cuml.model_selection._split import train_test_split 
```

Comment:
Since I'm paranoid and really want to make sure this PR is truly non-breaking, let's bring in  `_stratify_split` and `_approximate_mode` too.